Sources/CgRPC/shim/channel.c

```diff
@@ -47,7 +49,14 @@ cgrpc_channel *cgrpc_channel_create_secure(const char *address,
   channel_args.args = args;
   channel_args.num_args = num_args;
 
-  grpc_channel_credentials *creds = grpc_ssl_credentials_create(pem_root_certs, NULL, NULL);
+  grpc_ssl_pem_key_cert_pair *client_credentials = malloc(sizeof (struct grpc_ssl_pem_key_cert_pair));
+  client_credentials->cert_chain = client_certs;
+  client_credentials->private_key = client_private_key;
+  if (client_certs == NULL || client_private_key == NULL) {
+      client_credentials = NULL;
+  }
```

Comment:
Is there a more elegant way to do this? Also, do I need to `free(client_credentials)` after setting it to `NULL`?

Reply:
This is indeed leaking memory. Please look at https://github.com/grpc/grpc-swift/blob/38b66c346c007cbf52b7304a9ceddf907217d9a2/Sources/CgRPC/shim/server.c#L42 on how this should be done. You could have something like the following:

```
grpc_ssl_pem_key_cert_pair client_credentials;
grpc_ssl_pem_key_cert_pair *client_credentials_pointer = NULL;
if (client_certs != NULL && client_private_key != NULL) {
  client_credentials.cert_chain = client_certs;
  client_credentials.private_key = client_private_key;
  client_credentials_pointer = &client_credentials;
}
grpc_channel_credentials *creds = grpc_ssl_credentials_create(pem_root_certs, client_credentials_pointer, NULL);
```

Reply:
Also, please use 2 spaces for indentation. (Run `make project` to change this in your xcodeproj.)